dashboard/app/helpers/teacher_application_helper.rb

```diff
@@ -9,8 +9,8 @@ def current_application
       find_by(user: current_user)
   end
 
-  def has_incomplete_application?
-    !!current_application && current_application.status == 'incomplete'
+  def has_incomplete_open_application?
+    !!current_application && current_application.status == 'incomplete' && !current_application.regional_partner&.are_apps_closed
```

Comment:
sorry if this is covered elsewhere, but should there be something in this logic about whether the incomplete, open app is in the current application year?

Reply:
Yep, the tests document this -- we search for apps in the current application year, and one test has a user with both an old app and a new app (and only the new app is returned) and another test only has an old app (and no app is returned)